torch/_inductor/codecache.py

```diff
@@ -2004,12 +2004,14 @@ def _worker_compile(
     device_interface = get_interface_for_device(device.type)
     device_interface.Worker.set_device(device.index)
     kernel = TritonCodeCache.load(kernel_name, source_code)
-    kernel.precompile(warm_cache_only_with_cc=cc)
+    if hasattr(kernel, "precompile"):
```

Comment:
@jansel It looks like all the existing kernels are CachingAutotuner but in my case I end up with JitFunction from triton. I assume in order to support `@triton.autotune` I'm gonna need to make this work with cachingautotuner but for the time being is this fine?

Reply:
We should not use JitFunction from Triton, we want parallel ahead of time compiles.

You should be able to do something similar to:
https://github.com/pytorch/pytorch/blob/c84c86f018aea9aa9065ba3d57be1fc9fec51f5a/torch/_inductor/triton_heuristics.py#L1115-L1125

And put a `@template` above the generated Triton kernel.  You will need to generate proper `meta`.